### PR TITLE
Set up basic application structure.

### LINKED ISSUE #1: TCR: rails app plumbing
(This could (should?) spawn separate tickets)

- [x] github repo
- [ ] new rails app generator with api option
  - [ ] remove test directory and other unneeded dirs
- [ ] rspec generator
  - [ ] remove test directories we never use
- [ ] travis
- [ ] coveralls
- [ ] gemnasium
- [ ] badges in repo’s README for above tools
- [ ] version badge via fury.io in repo's README, use github release badge

#### From creating github repo:

image

### PR DESCRIPTION
**Pull request by tingulfsen**
Fixes #1 

This PR:

* Creates a new Rails application with the --api flag
* Sets Postgres to be the database
* Adds Travis
* Adds Coveralls
* Creates a basic README with the usual badges.
* Adds the Gems that I expect we'll need.

### CODE REVIEW

**Review by tallenaz — COMMENTED** [review]
Do we want to keep the Action Cable components?

**Comment by tingulfsen** [comment]
Good point - I believe I've removed the leftover ActionCable stuff. It seems unlikely we'll ever need it.

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+gem 'puma', '~> 3.7'
+gem 'rails', '~> 5.1'
+
+
+group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug', platforms: [:mri]
+  # Call 'binding.pry' anywhere in the code to stop execution and get a pry console
+  gem 'rspec-rails', '~> 3.6'
+  gem 'rails-controller-testing'
```
Comment (ndushay) [inline Gemfile]: add comment:  rails-controller-testing # allows use of assigns in tests

or delete gem until we need it?

File: README.md [hunk 2]
```diff
@@ -1 +1,9 @@
+[![Build Status](https://travis-ci.org/sul-dlss/trusted_checksum_repository.svg?branch=master)](https://travis-ci.org/sul-dlss/trusted_checksum_repository)
+[![Coverage Status](https://coveralls.io/repos/github/sul-dlss/trusted_checksum_repository/badge.svg)](https://coveralls.io/github/sul-dlss/trusted_checksum_repository)
+[![Dependency Status](https://gemnasium.com/badges/github.com/sul-dlss/trusted_checksum_repository.svg)](https://gemnasium.com/github.com/sul-dlss/trusted_checksum_repository)
+[![GitHub version](https://badge.fury.io/gh/sul-dlss%2Ftrusted_checksum_repository.svg)](https://badge.fury.io/gh/sul-dlss%2Ftrusted_checksum_repository)
+
+
 # README
+
+A Rails API for storing checksums. Used by the [Preservation Catalog](https://github.com/sul-dlss/preservation_core_catalog) as an external and restricted data store.
```
Comment (ndushay) [inline README.md]: maybe change line 9 to something like:

Trusted Checksum Repository is a Rails app for independently storing preservation object checksums.  Preservation Audit checks will use TCR to perform checksum validation on preserved objects.

File: Rakefile [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Add your own tasks in files placed in lib/tasks ending in .rake,
+# for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
+
+require_relative 'config/application'
+
+Rails.application.load_tasks
```
Comment (ndushay) [inline Rakefile:6]: I think we might like an explicit default rake task.

Would we want something like the following to make rubocop avail as a rake task?

```ruby
require 'rubocop/rake_task'
RuboCop::RakeTask.new
```

Would we want the default task to run spec, then rubocop?

```
task default: [:spec, :rubocop]
```

I believe for preservation_catalog, we had to write one no-op test so travis would be happy from the get go.  (That's probably how the controller confusion started)

Will we need to have travis do any postgres setup, or is this premature?

```
task :travis_setup_postgres do
  sh("psql -U postgres -f db/scripts/tcr_test_setup.sql")
end
```

File: config/database.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+development:
+  <<: *default
+  database: tcr
+  username: tcr
+
+test:
+  <<: *default
+  database: tcr
+  username: tcr
```
Comment (ndushay) [inline config/database.yml]: isn't this where we want tcr_test, just like pcc_test?

File: config/environments/development.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      'Cache-Control' => "public, max-age=#{2.days.seconds.to_i}"
+    }
+  else
+    config.action_controller.perform_caching = false
+
+    config.cache_store = :null_store
+  end
+
+  # Don't care if the mailer can't send.
+  config.action_mailer.raise_delivery_errors = false
```
Comment (ndushay) [inline config/environments/development.rb]: check preservation_catalog:  I think we can comment out a bunch of the config options pertaining to pieces we aren't using.  Ditto for production and test environments

File: config/application.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+# Pick the frameworks you want:
+require "active_model/railtie"
+require "active_job/railtie"
+require "active_record/railtie"
+require "action_controller/railtie"
+require "action_mailer/railtie"
+require "action_view/railtie"
+# require "sprockets/railtie"
+require "rails/test_unit/railtie"
+
```
Comment (ndushay) [inline config/application.rb]: for the above, can we comment out all the stuff we don't want?  This is all we have for preservation_catalog:

```
require 'rails'
require 'active_record/railtie'
require 'action_controller/railtie'
```

loading all these things is one of the ways spinning up the app takes so long.  I am even wondering if we have to include "rails" as a stand alone require.

**Review by ndushay — CHANGES_REQUESTED** [review]
Great start!  I suggested a number of tweaks.

File: Rakefile (hunk 3 above)
Comment (tingulfsen) [reply Rakefile:6]: Created default Rake task, and created #5

**Comment by tingulfsen** [comment]
@ndushay I think I've addressed your concerns above.

File: .rubocop_todo.yml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+# Cop supports --auto-correct.
+Layout/TrailingWhitespace:
+  Exclude:
+    - 'config/environments/production.rb'
+
+# Offense count: 25
+# Configuration parameters: AllowHeredoc, AllowURI, URISchemes, IgnoreCopDirectives, IgnoredPatterns.
+# URISchemes: http, https
+Metrics/LineLength:
+  Max: 130
```
Comment (ndushay) [inline .rubocop_todo.yml:48]: we're already up to line length 130?

File: .rubocop_todo.yml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+Style/PercentLiteralDelimiters:
+  Exclude:
+    - 'config/spring.rb'
+
+# Offense count: 11
+# Cop supports --auto-correct.
+# Configuration parameters: EnforcedStyle, SupportedStyles, ConsistentQuotesInMultiline.
+# SupportedStyles: single_quotes, double_quotes
+Style/StringLiterals:
+  Exclude:
```
Comment (ndushay) [inline .rubocop_todo.yml:83]: this is such a time wasting cop, we should just turn it off in .rubocop.yml

File: README.md [hunk 9]
```diff
@@ -1 +1,9 @@
+[![Build Status](https://travis-ci.org/sul-dlss/trusted_checksum_repository.svg?branch=master)](https://travis-ci.org/sul-dlss/trusted_checksum_repository)
+[![Coverage Status](https://coveralls.io/repos/github/sul-dlss/trusted_checksum_repository/badge.svg)](https://coveralls.io/github/sul-dlss/trusted_checksum_repository)
+[![Dependency Status](https://gemnasium.com/badges/github.com/sul-dlss/trusted_checksum_repository.svg)](https://gemnasium.com/github.com/sul-dlss/trusted_checksum_repository)
+[![GitHub version](https://badge.fury.io/gh/sul-dlss%2Ftrusted_checksum_repository.svg)](https://badge.fury.io/gh/sul-dlss%2Ftrusted_checksum_repository)
+
+
 # README
+
+Trusted Checksum Repository is a Rails app for independently storing preservation object checksums. The [Preservation Catalog](https://github.com/sul-dlss/preservation_core_catalog) uses the TCR to perform checksum validation on preserved objects.
```
Comment (ndushay) [inline README.md]: I'm not actually sure the PC will be calling the TCR directly;  that's what I was objecting to earlier.

**Review by ndushay — CHANGES_REQUESTED** [review]
Looks better, thanks!   Now that we have rubocop (I *knew* I was forgetting something in the ticket), I'm wondering if we want to fix any code and/or move some rubocop rule changes to `.rubocop.yml`

Also, I'm not yet convinced that the Preservation_catalog code will be calling the TCR code directly ... that was part of my objection to the README text before.

I could let the rubocop stuff go for now, but would prefer the README to say less rather than imply something we're not yet sure of.

File: .rubocop_todo.yml (hunk 7 above)
Comment (tingulfsen) [reply .rubocop_todo.yml:48]: There is no application code here yet - this is rubocop complaining about auto-generated stuff that Rails makes. I am not going to edit that auto-generated stuff now.

**Review by ndushay — APPROVED** [review]

**Pull request merged**